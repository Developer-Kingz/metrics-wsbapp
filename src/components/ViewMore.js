/* eslint-disable */
import React from 'react';
import { useSelector } from 'react-redux';
import { useParams, NavLink } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowLeft } from '@fortawesome/free-solid-svg-icons';
import NavBar from './NavBar';

const ViewMore = () => {
  const { name } = useParams();

  const country = useSelector((state) => state.data.data);
  const details = country.find((cont) => cont.name.common === name);

  return (
    <>
      <NavBar />
      <div className="viewmore">
        <NavLink to="/metrics-wsbapp" className="nav"><FontAwesomeIcon className="arr" icon={faArrowLeft} /></NavLink>
        <div className="details-cont">
          <img src={details.flags.svg} alt="" className="cont-flag" />

          <div className="details-cont-inner">
              <div>
                <p>Country Name:</p>
                {details.name.common}
              </div>
              <div>
                <p>Capital:</p>
                {details.capital}
              </div>
          
              <div>
                <p>Region:</p>
                {details.region}
              </div>
              <div>
                <p>Countries Population:</p>
                {details.population}
              </div>
              <div>
                <p>Subregion:</p>
                {details.subregion}
              </div>
              <div> <p>TimeZone:</p> {details.timezones}</div>
            </div>
        </div>
      </div>
    </>
  );
};

export default ViewMore;
