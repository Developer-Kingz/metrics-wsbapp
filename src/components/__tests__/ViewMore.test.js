import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import ViewMore from '../ViewMore';

const mockStore = configureMockStore([]);

const initialState = {
  data: {
    data: [
      {
        name: 'Test Country',
        flags: 'test-flag-url',
        officialName: 'Official Test Country',
        capital: 'Test Capital',
        continent: 'Test Continent',
        population: 'Test Population',
        area: 'Test Area',
        languages: { ukr: 'spain' },
        timezone: ['Test Timezone'],
      },
    ],
  },
};

describe('ViewMore Component', () => { 
  it('renders correctly', () => {
    const store = mockStore(initialState);

    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <ViewMore />
        </BrowserRouter>
      </Provider>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
