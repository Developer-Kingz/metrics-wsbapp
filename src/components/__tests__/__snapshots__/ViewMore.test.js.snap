// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewMore Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <div
      class="nav-inner"
    >
      <img
        alt=""
        class="nav-img"
        src="https://cdn.britannica.com/94/75394-050-C3F58654/World-map-Continents-Oceans-Seas.jpg"
      />
      <h4>
        Search Countries
      </h4>
    </div>
    <input
      class="search-country"
      placeholder="Search Country"
      type="text"
      value=""
    />
  </div>
  <div
    class="viewmore"
  >
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-left arr"
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
          fill="currentColor"
        />
      </svg>
    </a>
    <div
      class="details-cont"
    >
      <img
        alt=""
        class="cont-flag"
      />
      <div
        class="details-cont-inner"
      >
        <div
          class="cont-in"
        >
          <h1>
            Country Name: 
          </h1>
          <h3>
            Capital: Test Capital
          </h3>
        </div>
        <div
          class="region"
        >
          <h4>
            Region: 
          </h4>
          <h4>
            Countries Population: Test Population
          </h4>
          <h4>
            Language: spain
          </h4>
          <h4 />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
